plan/plan.go

```diff
@@ -240,6 +240,10 @@ func (p *Plan) Calculate() *Plan {
 
 				if ownersMatch {
 					changes.Create = append(changes.Create, creates...)
+				} else {
+					for _, current := range row.current {
+						log.Debugf(`Skipping endpoint %v because owner id does not match for one or more items to create, found: "%s", required: "%s"`, current, current.Labels[endpoint.OwnerLabelKey], p.OwnerID)
+					}
```

Comment:
If you do something like this, it would avoid a useless for loop for everybody that doesn't configure the logger at the debug level (note: requires proper indentation). 

```suggestion
if log.GetLevel() == log.DebugLevel {
	for _, current := range row.current {
		log.Debugf(`Skipping endpoint %v because owner id does not match for one or more items to create, found: "%s", required: "%s"`, current, current.Labels[endpoint.OwnerLabelKey], p.OwnerID)
	}
}
```